cmd/konvoy-image-wrapper/cmd/wrapper.go

```diff
@@ -150,6 +152,11 @@ func (r *Runner) setAWSEnv() error {
 		}
 	}
 
+	_, found := os.LookupEnv(envAWSCredentialsFile)
+	if !found {
+		// fall-back to default location for aws credentials file
+		os.Setenv(envAWSCredentialsFile, filepath.Join(r.usr.HomeDir, ".aws", "credentials"))
```

Comment:
When user's homedir mount was removed, the aws creds from `credentials` stopped working. So defaulting `AWS_SHARED_CREDENTIALS_FILE` to ~/.aws/credentials as per the variable's documentation given at https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

Reply:
Do we have to review this for SSH agent socket or other cloud providers?

Reply:
I can make ssh agent socket readonly and test it out. I am not sure if packer or KIB writes to this socket.
The wrapper mounts GCP credentials file to the container. Mounting files pointed by credentials env variable was moved to a common function [mountFileEnv](https://github.com/mesosphere/konvoy-image-builder/pull/589/files#diff-d8016cb551cfda4bbd3bc9124263f9003367a4cf5a6d144b96c928faec12f3aaR217) in last refactor. so above fix will apply to all the cloud provider's credentials.

Reply:
Sounds good. Let me know how it goes.

Reply:
mounted `SSH_AUTH_SOCK` as readonly: https://github.com/mesosphere/konvoy-image-builder/pull/589/commits/c4368fbfd316b22223e920f3c92f0da0b7d23ed6